modules/sor/utils/data.ts

```diff
@@ -18,15 +19,56 @@ export type BufferPoolData = {
     unwrapRate: bigint;
 };
 
-const cache = new Cache<string, { pools: PrismaPoolAndHookWithDynamic[]; bufferPools: BufferPoolData[] }>();
+export type SORDbPool = PrismaPoolAndHookWithDynamic;
+
+const cache = new Cache<string, { pools: SORDbPool[]; bufferPools: BufferPoolData[] }>();
 const SOR_POOLS_CACHE_KEY = `sor:pools`;
 
 export async function getBasePoolsFromDb(
     chain: Chain,
     protocolVersion: number,
     considerPoolsWithHooks: boolean,
     poolIds?: string[],
-): Promise<{ pools: PrismaPoolAndHookWithDynamic[]; bufferPools: BufferPoolData[] }> {
+): Promise<{ pools: SORDbPool[]; bufferPools: BufferPoolData[] }> {
+    const cacheKey = `${SOR_POOLS_CACHE_KEY}:${chain}`;
+
+    let cached = cache.get(cacheKey);
+
+    if (!cached) {
+        // get pools
+        console.time('SOR:getpools');
+        const pools = await getPools(chain);
+        console.timeEnd('SOR:getpools');
+        const bufferPools = await getBufferPoolsFromDBPools(pools, chain);
+        cached = cache.put(cacheKey, { pools, bufferPools }, 10 * 1000); // cache for 10s
```

Comment:
Could further optimize this and only `getBufferPoolsFromDBPools` if its a v3 run. Also means we need two caches, one for v2 and one for v3 runs.